pkg/bmc/idrac_virtualmedia.go

```diff
@@ -86,3 +88,53 @@ func (a *redfishiDracVirtualMediaAccessDetails) RAIDInterface() string {
 func (a *redfishiDracVirtualMediaAccessDetails) VendorInterface() string {
 	return "no-vendor"
 }
+
+func (a *redfishiDracVirtualMediaAccessDetails) ValidateBootInterface(bmcCreds Credentials) (bool, error) {
+	var hasVirtualMedia, err = a.validateVirtualMedia(bmcCreds)
+	return hasVirtualMedia, err
+}
+
+func (a *redfishiDracVirtualMediaAccessDetails) validateVirtualMedia(bmcCreds Credentials) (bool, error) {
+	bmcConfig := gofish.ClientConfig{
+		Endpoint: getRedfishAddress(a.bmcType, a.host),
+		Username: bmcCreds.Username,
+		Password: bmcCreds.Password,
+		Insecure: !a.disableCertificateVerification,
+	}
+
+	client, err := gofish.Connect(bmcConfig)
+	if err != nil {
+		panic(err)
+	}
+	defer client.Logout()
+
+	service := client.Service
+
+	managers, err := service.Managers()
+	if err != nil {
+		panic(err)
+	}
+
+	for _, manager := range managers {
+		vmediaList, err := manager.VirtualMedia()
+		if err != nil {
+			return false, err
+		}
+		if vmediaList == nil {
+			return false, &BootInterfaceValidationError{message: "BMC doesn't have any members for VirtualMedia"}
+		}
+
+		for _, vmedia := range vmediaList {
+			// Maybe we just need to check in CD
+			if vmedia.Entity.ID == "CD" {
```

Comment:
Let us check MediaTypes, not ID. I'm not sure IDs are guaranteed to stay the same forever, while MediaTypes is an enumeration.

Reply:
In go it is more usual to perform a test to let you either continue or return early, to avoid excessively deep nesting and to make the "happy path" through the code clear. So in this case I think you want to reverse the test and `continue` instead of indenting the rest of the checks inside this if block.

Reply:
Done